src/task-actions.ts

```diff
@@ -164,7 +165,12 @@ export async function starknetCompileCairo1Action(
     args: TaskArguments,
     hre: HardhatRuntimeEnvironment
 ) {
-    const binDirPath = getCompilerBinDir(args, hre.config.starknet);
+    let binDirPath = getCompilerBinDir(args, hre.config.starknet);
+    if (!binDirPath) {
+        const downloader = new CairoCompilerDownloader(hre.config.paths.root, hre.config.starknet);
+        await downloader.handleCompilerDownload();
+        binDirPath = downloader.getBinDirPath();
```

Comment:
Ideally this would be just one call. I.e. some get method should ensure the compiler version is downloaded and return the path after that.